akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -174,8 +303,123 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
         case (None, None) ⇒ throw new IllegalStateException(s"missing value for $key")
       }
     }
+    val removedKeysIterator = (tombstoneVals.keys ++ that.tombstoneVals.keys).iterator
+    var removedValues = Map.empty[A, B]
+    removedKeysIterator.foreach { key ⇒
+      (this.tombstoneVals.get(key), that.tombstoneVals.get(key)) match {
+        case (Some(thisValue), Some(thatValue)) ⇒
+          if (thisValue.getClass != thatValue.getClass) {
+            val errMsg = s"Wrong type for merging [$key] in [${getClass.getName}], existing type " +
+              s"[${thisValue.getClass.getName}], got [${thatValue.getClass.getName}]"
+            throw new IllegalArgumentException(errMsg)
+          }
+          val removedValue = thisValue.merge(thatValue.asInstanceOf[thisValue.T]).asInstanceOf[B]
+          removedValues = removedValues.updated(key, removedValue)
+        case (Some(thisValue), None) ⇒
+          removedValues = removedValues.updated(key, thisValue)
+        case (None, Some(thatValue)) ⇒
+          removedValues = removedValues.updated(key, thatValue)
+        case (None, None) ⇒ throw new IllegalStateException(s"missing value for $key")
+      }
+    }
+    // tombstone the values for removed keys
```

Comment:
That would be complicated, since the cluster membership is dynamic. No, that is the wrong avenue. Could we get away without tombstones altogether if we don't use deltas for `put` and `remove`?